### PR TITLE
"Final" PR to add all changes for weekly checks

### PR DESCRIPTION
**Pull request by benkrikler**
This PR implements the weekly_analyzer used for the weekly checks of the jet sums.  
Should be merged after #76.

### CODE REVIEW

File: cmsl1t/plotting/resolution.py [hunk 1]
```diff
@@ -0,0 +1,104 @@
...
+
+            # Save canvas to file
+            name = self.filename_format.format(pileup="all")
+            self.save_canvas(canvas, name + suffix)
+
+    def _is_consistent(self, new):
+        """
+        Check the two plotters are the consistent, so same binning and same axis names
+        """
+        return (self.pileup_bins.bins == new.pileup_bins.bins) and \
```
Comment (kreczko) [inline cmsl1t/plotting/resolution.py]: same comment as for the previous PR
&rarr; use `all`

File: cmsl1t/plotting/resolution_vs_X.py [hunk 2]
```diff
@@ -0,0 +1,84 @@
...
+            # Add labels
+            label_canvas()
+
+            # Save canvas to file
+            name = self.filename_format.format(pileup=pileup)
+            self.save_canvas(canvas, name + suffix)
+
+    def _is_consistent(self, new):
+        """
+        Check the two plotters are the consistent, so same binning and same axis names
```
Comment (kreczko) [inline cmsl1t/plotting/resolution_vs_X.py]: &rarr; use `all`

File: cmsl1t/recalc/resolution.py [hunk 3]
```diff
@@ -0,0 +1,67 @@
+import numpy as np
+from exceptions import RuntimeError
+from math import pi, degrees
+twopi = 2 * pi
+
+
+def get_resolution_function(resolution_type):
+    if resolution_type.lower() == "energy":
```
Comment (kreczko) [inline cmsl1t/recalc/resolution.py]: might be worth to cache `resolution_type.lower()`, i.e `rtype = resolution_type.lower()` and use that for comparison #speed

File: cmsl1t/plotting/resolution.py (hunk 1 above)
Comment (benkrikler) [reply cmsl1t/plotting/resolution.py]: These files were added in #76, I'm not sure why this hasn't updated the diff, but I'll merge `weekly-checks` back into this and push to update things.

Comment (benkrikler) [reply cmsl1t/plotting/resolution.py]: Merged in the updated `weekly-checks` branch to this and now the diff looks as I'd expected.

**Pull request merged**